jwst/pixel_replace/pixel_replace.py

```diff
@@ -417,28 +515,50 @@ def mingrad(self, model):
+
             left_err, right_err = inerr[yindx[ii], xindx[ii] - 1], inerr[yindx[ii], xindx[ii] + 1]
             top_err, bottom_err = inerr[yindx[ii] - 1, xindx[ii]], inerr[yindx[ii] + 1, xindx[ii]]
 
+            left_var_p, right_var_p = in_var_p[yindx[ii], xindx[ii] - 1], in_var_p[yindx[ii], xindx[ii] + 1]
+            top_var_p, bottom_var_p = in_var_p[yindx[ii] - 1, xindx[ii]], in_var_p[yindx[ii] + 1, xindx[ii]]
+
+            left_var_r, right_var_r = in_var_r[yindx[ii], xindx[ii] - 1], in_var_r[yindx[ii], xindx[ii] + 1]
+            top_var_r, bottom_var_r = in_var_r[yindx[ii] - 1, xindx[ii]], in_var_r[yindx[ii] + 1, xindx[ii]]
+
+            left_var_f, right_var_f = in_var_f[yindx[ii], xindx[ii] - 1], in_var_f[yindx[ii], xindx[ii] + 1]
+            top_var_f, bottom_var_f = in_var_f[yindx[ii] - 1, xindx[ii]], in_var_f[yindx[ii] + 1, xindx[ii]]
+
             # Compute absolute difference (slope) and average value in each direction (may be NaN)
             diffs = np.array([np.abs(left_data - right_data), np.abs(top_data - bottom_data)])
             interp_data = np.array([(left_data + right_data) / 2., (top_data + bottom_data) / 2.])
             interp_err = np.array([(left_err + right_err) / 2., (top_err + bottom_err) / 2.])
+            interp_var_p = np.array([(left_var_p + right_var_p) / 2., (top_var_p + bottom_var_p) / 2.])
+            interp_var_r = np.array([(left_var_r + right_var_r) / 2., (top_var_r + bottom_var_r) / 2.])
+            interp_var_f = np.array([(left_var_f + right_var_f) / 2., (top_var_f + bottom_var_f) / 2.])
 
             # Replace with the value from the lowest absolute slope estimator that was not NaN
             try:
                 indmin = np.nanargmin(diffs)
                 newdata[yindx[ii], xindx[ii]] = interp_data[indmin]
                 newerr[yindx[ii], xindx[ii]] = interp_err[indmin]
-                # If original pixel was in the science array set DQ accordingly
-                if ((indq[yindx[ii], xindx[ii]] & self.NON_SCIENCE) == 0):
-                    newdq[yindx[ii], xindx[ii]] = self.FLUX_ESTIMATED
-                # If original pixel was in non-science region set DQ accordingly
-                else:
-                    newdq[yindx[ii], xindx[ii]] = self.FLUX_ESTIMATED + self.NON_SCIENCE + self.DO_NOT_USE
+
+                # Square the interpolated errors back into variance
+                new_var_p[yindx[ii], xindx[ii]] = interp_var_p[indmin] ** 2
+                new_var_r[yindx[ii], xindx[ii]] = interp_var_r[indmin] ** 2
+                new_var_f[yindx[ii], xindx[ii]] = interp_var_f[indmin] ** 2
+
+                # If original pixel was in the science array, remove
+                # the DO_NOT_USE flag
+                if ((indq[yindx[ii], xindx[ii]] & self.DO_NOT_USE)
+                        and not (indq[yindx[ii], xindx[ii]] & self.NON_SCIENCE)):
+                    newdq[yindx[ii], xindx[ii]] -= self.DO_NOT_USE
```

Comment:
I modified this section to preserve any original flags in the replaced pixel, instead of replacing them with FLUX_ESTIMATED only, to bring this algorithm in line with what fit_profile does.